core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java

```diff
@@ -423,22 +421,13 @@ public int parseInto(ReadWritablePeriod period, String text, int position, Local
                     }
                 }
                 else if (parsePos < 0) {
-                    parsePos = ~parsePos;
-                    if (parsePos > bestInvalidPos) {
-                        bestInvalidPos = parsePos;
-                    }
```

Comment:
I think there is no reason to continue trying to parse when even one of parsers fails in this use case.